agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -126,8 +166,28 @@ CompletableResultCode internalSendByConnectionString(
     return telemetryPipeline.send(byteBuffers, connectionString, listener);
   }
 
-  List<ByteBuffer> encode(List<TelemetryItem> telemetryItems) throws IOException {
+  private static TelemetryItem createOtelResourceMetric(
+      Map<String, String> existingTags, String connectionString) {
+    MetricTelemetryBuilder builder = MetricTelemetryBuilder.create(_OTELRESOURCE_, 0);
+    builder.setConnectionString(connectionString);
+    builder.addTag(
+        ContextTagKeys.AI_CLOUD_ROLE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE.toString()));
+    builder.addTag(
+        ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString()));
+    builder.addTag(
+        ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString(),
+        existingTags.get(ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString()));
+
+    // add attributes from OTEL_RESOURCE_ATTRIBUTES
+    for (Map.Entry<String, String> entry : OTEL_RESOURCE_ATTRIBUTES.entrySet()) {
+      builder.addProperty(entry.getKey(), entry.getValue());
```

Comment:
similar to https://github.com/open-telemetry/opentelemetry-java/blob/c201396f57b384436de63617af29424cb4a7e035/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java#L74-L82:

```suggestion
      // Attributes specified via otel.resource.attributes follow the W3C Baggage spec and
      // characters outside the baggage-octet range are percent encoded
      // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable
      builder.addProperty(entry.getKey(), URLDecoder.decode(entry.getValue(), StandardCharsets.UTF_8.displayName()));
```